tsdb/engine/tsm1/engine.go

```diff
@@ -1331,7 +1331,9 @@ func (e *Engine) onFileStoreReplace(newFiles []TSMFile) {
 
 	// load metadata from the Cache
 	e.Cache.ApplyEntryFn(func(key []byte, entry *entry) error {
+		entry.mu.RLock()
```

Comment:
Instead of touching the entries lock from the engine, can you change this to have a `entry.InfluxQLType()` instead that keeps the locking private to the `entry`?

Reply:
Sure.